app/controllers/apps_controller.rb

```diff
@@ -0,0 +1,37 @@
+class AppsController < ApplicationController
+
+  def new
+    @favorite_pets = @favorite.favorite_pets
+  end
+
+  def create
+    @app = App.create(app_params)
+
+    if @app.save
```

Comment:
if all parameters are present create the PetApp join table, then delete pet from fav list